framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -0,0 +1,188 @@
+const _ = require('lodash');
+
+const transferAsset = raw => {
+	if (raw.tf_data) {
+		// This line will throw if there is an error
+		const data = raw.tf_data.toString('utf8');
+		return { data };
+	}
+	return null;
+};
+
+const signatureAsset = raw => {
+	if (!raw.s_publicKey) {
+		return null;
+	}
+	const signature = {
+		transactionId: raw.t_id,
+		publicKey: raw.s_publicKey,
+	};
+
+	return { signature };
+};
+
+const delegateAsset = raw => {
+	if (!raw.d_username) {
+		return null;
+	}
+	const delegate = {
+		username: raw.d_username,
+		publicKey: raw.t_senderPublicKey,
+		address: raw.t_senderId,
+	};
+
+	return { delegate };
+};
+
+const voteAsset = raw => {
+	if (!raw.v_votes) {
+		return null;
+	}
+	const votes = raw.v_votes.split(',');
+
+	return { votes };
+};
+
+const multiAsset = raw => {
+	if (!raw.m_keysgroup) {
+		return null;
+	}
+	const multisignature = {
+		min: raw.m_min,
+		lifetime: raw.m_lifetime,
+	};
+
+	if (typeof raw.m_keysgroup === 'string') {
+		multisignature.keysgroup = raw.m_keysgroup.split(',');
+	} else {
+		multisignature.keysgroup = [];
+	}
+
+	return { multisignature };
+};
+
+const dappAsset = raw => {
+	if (!raw.dapp_name) {
+		return null;
+	}
+	const dapp = {
+		name: raw.dapp_name,
+		description: raw.dapp_description,
+		tags: raw.dapp_tags,
+		type: raw.dapp_type,
+		link: raw.dapp_link,
+		category: raw.dapp_category,
+		icon: raw.dapp_icon,
+	};
+
+	return { dapp };
+};
+
+const inTransferAsset = raw => {
+	if (!raw.in_dappId) {
+		return null;
+	}
+	const inTransfer = {
+		dappId: raw.in_dappId,
+	};
+
+	return { inTransfer };
+};
+
+const outTransferAsset = raw => {
+	if (!raw.ot_dappId) {
+		return null;
+	}
+	const outTransfer = {
+		dappId: raw.ot_dappId,
+		transactionId: raw.ot_outTransactionId,
+	};
+
+	return { outTransfer };
+};
+
+class Transaction {
```

Comment:
I think the name of the class is not self-describing enough. My proposal is: `TransactionInitializer` or something similar. Also, this file is exporting only this class while having a different name.

Reply:
Originally, we thought this would be temporal class, but as far as chain module concern, there is only one Transaction in general way, and i think it's a good name?

Reply:
What I find misleading is the mismatching names of the file and the content of it. The file is named `init_transaction` while containing a class named `Transaction`. I think naming it `transaction` is more consistent if you want to keep the name of the class untouched, however afaict, this class is responsible for instantiating other different classes that hold different transaction implementations so other names such as `TransactionInitializer` I find them more suitable for the purpose of the class.

Reply:
yea the file and the function is mismatching and confusing.
We should definitely fix that. Let's open another issue for this

Reply:
Issue: https://github.com/LiskHQ/lisk-sdk/issues/3447